components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/utils/SecondaryUserStoreConfigurator.java

```diff
@@ -144,11 +138,48 @@ private void initializeKeyStore() throws IdentityUserStoreMgtException {
      */
     public String encryptPlainText(String plainText) throws IdentityUserStoreMgtException {
 
-        if (cipher == null) {
-            initializeKeyStore();
+        boolean isInternalKeyStoreEncryptionEnabled = false;
+        boolean isSymmetricKeyEncryptionEnabled = false;
+        ServerConfigurationService config =
+                UserStoreConfigComponent.getServerConfigurationService();
+        if (config != null) {
+            String encryptionKeyStore = config.getFirstProperty(ENCRYPTION_KEYSTORE);
+
+            if (INTERNAL_KEYSTORE.equalsIgnoreCase(encryptionKeyStore)) {
+                isInternalKeyStoreEncryptionEnabled = true;
+            }
+            String cryptoProvider = config.getFirstProperty(CRYPTO_PROVIDER);
+            if (SYMMETRIC_KEY_CRYPTO_PROVIDER.equalsIgnoreCase(cryptoProvider)) {
+                isSymmetricKeyEncryptionEnabled = true;
+            }
         }
 
+        if (isInternalKeyStoreEncryptionEnabled && isSymmetricKeyEncryptionEnabled) {
+
+            throw new IdentityUserStoreMgtException(String.format("Userstore encryption cazn not be supported due to " +
+                    "conflicting configurations: '%s' and '%s'. When using internal keystore, assymetric crypto " +
+                    "provider should be used.", INTERNAL_KEYSTORE, SYMMETRIC_KEY_CRYPTO_PROVIDER));
+        } else if (isInternalKeyStoreEncryptionEnabled || isSymmetricKeyEncryptionEnabled) {
+
+            try {
+                return CryptoUtil.getDefaultCryptoUtil().encryptAndBase64Encode(plainText.getBytes());
+            } catch (CryptoException e) {
+                String errorMessage = "Error while encrypting the plain text using internal keystore.";
+                throw new IdentityUserStoreMgtException(errorMessage, e);
+            }
+        } else {
+            return encryptWithPrimaryKeyStore(config, plainText);
+        }
+
+    }
```

Comment:
What's the difference of this and the one in UserCoreUtil.
Isn't same logic is duplicated in both places

Reply:
Those 2 classes are in 2 different OSGI bundles.
UserStoreUtil - org.wso2.carbon.identity.user.store.configuration.deployer
SecondaryUserStoreConfigurator. - org.wso2.carbon.identity.user.store.configuration

Reply:
> What's the difference of this and the one in UserCoreUtil.
> Isn't same logic is duplicated in both places

Created a new git issue https://github.com/wso2/product-is/issues/7954 to track this